### DISCUSSION TITLE
Enabling SSL - how do I?

### DISCUSSION THREAD

**Discussion by RickDG2** [lbbrhzn/ocpp#591] [Q&A]
How do I enable SSL (for WSS://)?

In my specific case, the EVSE I am trying to test with (an Autel Maxicharger) doesn't currently support changing the OCPP settings.

There are multiple pre-sets, but you cannot select any of them.

However, it does default to **wss://gateway-eneprodus.autel.com:443/ws/webSocket?sn=xxxxx**

I can easily override DNS in my firewall for that FQDN and point it at my HA server.

But I don't see any support for configuring SSL in the ocpp integration, short of manually editing the configuration files in config/custom_components/ocpp

And even then, it's not clear to me if changing const.py:DEFAULT_SSL = False to True and restarting will be sufficient.

Plus I am pretty sure that my change would be overridden by the next update, but for testing, I could live with this for now.

I realize that while DNS is being overridden, I won't be getting any firmware updates from autel.com unless I occasionally disable the override, reboot the EVSE, make it check for firmware updates, etc., then re-enable the DNS override.

Do-able until they update the firmware to support the manual entry of OCPP configuration, which is supported by the iOS app, but always tells me "Current firmware version does not support this feature.  Please update to the latest firmware version." - which of course, I have the current firmware version installed.

Any help would be appreciated.

**Comment by drc38** [comment]
See https://github.com/lbbrhzn/ocpp/pull/506 for instructions noting the secure option has not been tested and may not work.
